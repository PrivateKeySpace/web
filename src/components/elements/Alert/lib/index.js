export { default as propTypes } from './propTypes'
export { default as defaultPropsSpec } from './defaultPropsSpec'
