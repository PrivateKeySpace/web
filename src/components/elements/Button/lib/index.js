export { default as mapProps } from './mapProps'
export { default as propTypes } from './propTypes'
