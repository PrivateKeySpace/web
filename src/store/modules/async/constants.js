export const STATUS_PENDING = 'pending'
export const STATUS_SUCCESS = 'success'
export const STATUS_FAILURE = 'failure'

export const METHOD_GET = 'GET'
export const METHOD_POST = 'POST'
export const METHOD_PATCH = 'PATCH'
export const METHOD_PUT = 'PUT'
