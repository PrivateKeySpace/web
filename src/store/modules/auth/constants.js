export const STORAGE_RECORD_KEY = 'auth'

// One hour TTL, subject to change
export const STORAGE_RECORD_TTL = 60 * 60 * 1000
