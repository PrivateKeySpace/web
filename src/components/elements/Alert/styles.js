import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../theme/constants'

const styles = StyleSheet.create({
  alert: {
    height: 48,
    fontSize: 12,
    backgroundColor: colors.white,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: colors.lightGray,
    borderLeftWidth: 3
  },
  danger: {
    borderColor: colors.danger,
    backgroundColor: colors.lightDanger
  },
  success: {
    borderColor: colors.success,
    backgroundColor: colors.lightSuccess
  },
  warning: {
    borderColor: colors.warning,
    backgroundColor: colors.lightWarning
  },
  info: {
    borderColor: colors.info,
    backgroundColor: colors.lightInfo
  },
  controls: {
    marginLeft: 'auto'
  }
})

export default styles
