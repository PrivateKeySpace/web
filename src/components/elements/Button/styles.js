import { StyleSheet } from 'aphrodite/no-important'
import { darken } from 'polished'
import { colors, variants } from '../../../theme/constants'

const buttonHeight = 38

const buttonStyle = {
  height: buttonHeight,
  borderRadius: buttonHeight,
  fontSize: 13,
  cursor: 'pointer',
  userSelect: 'none',
  outline: 'none',
  paddingLeft: 16,
  paddingRight: 16
}

export const defaultSx = StyleSheet.create({
  button: {
    ...buttonStyle,
    backgroundColor: colors.lightGray,
    border: 'none',
    color: colors.black
  },
  ...variants.reduce((acc, variant) => {
    acc[variant] = {
      color: colors.white,
      backgroundColor: colors[variant],
      ':hover': {
        backgroundColor: darken(0.1, colors[variant])
      },
      ':active': {
        backgroundColor: darken(0.15, colors[variant])
      }
    }
    return acc
  }, {})
})

export const outlineSx = StyleSheet.create({
  button: {
    ...buttonStyle,
    background: 'transparent',
    borderWidth: 2,
    borderStyle: 'solid',
    borderColor: colors.black,
    color: colors.black,
    paddingLeft: 14,
    paddingRight: 14
  },
  ...variants.reduce((acc, variant) => {
    acc[variant] = {
      borderColor: colors[variant],
      color: colors[variant],
      ':hover': {
        borderColor: darken(0.1, colors[variant]),
        backgroundColor: darken(0.1, colors[variant]),
        color: colors.white
      },
      ':active': {
        borderColor: darken(0.15, colors[variant]),
        backgroundColor: darken(0.15, colors[variant]),
        color: colors.white
      }
    }
    return acc
  }, {})
})

export const linkSx = StyleSheet.create({
  button: {
    ...buttonStyle,
    background: 'transparent',
    border: 'none',
    color: colors.black
  },
  ...variants.reduce((acc, variant) => {
    acc[variant] = {
      color: colors[variant],
      ':hover': {
        color: darken(0.1, colors[variant])
      },
      ':active': {
        color: darken(0.15, colors[variant])
      }
    }
    return acc
  }, {})
})
