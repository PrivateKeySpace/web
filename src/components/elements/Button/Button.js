import React from 'react'
import { css } from 'aphrodite/no-important'
import tx from '../../../theme/styles'
import { defaultSx, outlineSx, linkSx } from './styles'

function getSx (outline, link) {
  if (link) {
    return linkSx
  } else if (outline) {
    return outlineSx
  } else {
    return defaultSx
  }
}

function Button (props) {
  const { variant, children, outline, link, ...nativeProps } = props
  const sx = getSx(outline, link)

  return (
    <button
      {...nativeProps}
      className={css(tx.px4, tx.textTransformUppercase, tx.fontWeight500, tx.inlineFlex, tx.alignItemsCenter, tx.justifyContentCenter, sx.button, variant && sx[variant])}
    >
      {children}
    </button>
  )
}

export default Button
