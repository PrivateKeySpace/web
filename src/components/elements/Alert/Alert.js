import React from 'react'
import { css } from 'aphrodite/no-important'
import tx from '../../../theme/styles'
import sx from './styles'
import Button from '../Button'

function Alert (props) {
  const { text, variant } = props

  return (
    <div className={css(tx.width100, tx.fontWeight500, tx.mb1, tx.px3, tx.flex, tx.flexDirectionRow, tx.alignItemsCenter, sx.alert, variant && sx[variant])}>
      <div className={css(sx.text)}>{text}</div>
      <div className={css(sx.controls)}>
        <Button>Test</Button>
      </div>
    </div>
  )
}

export default Alert
