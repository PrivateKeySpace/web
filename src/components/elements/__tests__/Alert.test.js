import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import { variants } from '../../../theme/constants'
import Alert from '../Alert'
import Button from '../Button'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders alerts of all variants with no controls, does not crash and matches snapshot', () => {
  for (const variant of variants) {
    const alert = render(
      <Alert variant={variant} text='Test alert' />
    )
    expect(alert).toMatchSnapshot()
  }
})

it('renders alerts of all variants with multiple controls, does not crash and matches snapshot', () => {
  for (const variant of variants) {
    const alert = render(
      <Alert
        variant={variant}
        text='Test alert'
        controls={[
          <Button variant='danger'>Close</Button>,
          <Button variant='info'>Accept</Button>
        ]}
      />
    )
    expect(alert).toMatchSnapshot()
  }
})
