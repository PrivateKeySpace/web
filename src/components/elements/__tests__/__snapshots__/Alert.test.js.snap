// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders alerts of all variants with multiple controls, does not crash and matches snapshot 1`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-danger_a5o1zu"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-danger_o6beaq"
      >
        Close
      </button>
    </div>
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-info_ki9aj1"
      >
        Accept
      </button>
    </div>
  </div>
</div>
`;

exports[`renders alerts of all variants with multiple controls, does not crash and matches snapshot 2`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-success_1b63wt8"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-danger_o6beaq"
      >
        Close
      </button>
    </div>
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-info_ki9aj1"
      >
        Accept
      </button>
    </div>
  </div>
</div>
`;

exports[`renders alerts of all variants with multiple controls, does not crash and matches snapshot 3`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-warning_gn949f"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-danger_o6beaq"
      >
        Close
      </button>
    </div>
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-info_ki9aj1"
      >
        Accept
      </button>
    </div>
  </div>
</div>
`;

exports[`renders alerts of all variants with multiple controls, does not crash and matches snapshot 4`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-info_1acwq04"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-danger_o6beaq"
      >
        Close
      </button>
    </div>
    <div
      class="px1_17oj4c4"
    >
      <button
        class="px4_woxb04-o_O-textTransformUppercase_4efw5a-o_O-fontWeight500_gt7myn-o_O-inlineFlex_goo6eo-o_O-alignItemsCenter_1moixrt-o_O-justifyContentCenter_agrn11-o_O-button_1pvg0dr-o_O-info_ki9aj1"
      >
        Accept
      </button>
    </div>
  </div>
</div>
`;

exports[`renders alerts of all variants with no controls, does not crash and matches snapshot 1`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-danger_a5o1zu"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    
  </div>
</div>
`;

exports[`renders alerts of all variants with no controls, does not crash and matches snapshot 2`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-success_1b63wt8"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    
  </div>
</div>
`;

exports[`renders alerts of all variants with no controls, does not crash and matches snapshot 3`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-warning_gn949f"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    
  </div>
</div>
`;

exports[`renders alerts of all variants with no controls, does not crash and matches snapshot 4`] = `
<div
  class="width100_zdxht7-o_O-fontWeight500_gt7myn-o_O-mb1_1tpeo8l-o_O-px3_1rzpu44-o_O-flex_jro6t0-o_O-flexDirectionRow_qb00m9-o_O-alignItemsCenter_1moixrt-o_O-alert_ykn234-o_O-info_1acwq04"
>
  <div
    class=""
  >
    Test alert
  </div>
  <div
    class="flex_jro6t0-o_O-controls_ls0e43"
  >
    
  </div>
</div>
`;
